atlp-rwanda/knights-ecomm-fe
# #68 Implement vendor and admin order management feature

#### What does this PR do?
This PR implements order management functionalities for both vendors and admins. It allows them to view and manage orders placed by buyers and perform actions on these orders where vendor is required to fullfil the request from buyer, while admins  control the interactions between buyers and vendors.

#### Description of Task to be completed?
- implement UI for both vendor and admin order management including viewing all orders and single order.
- Apply styling to order management components using Tailwind CSS.
- Ensure responsive design for different screen sizes.
- Ensure search and pagination work smoothly 
- Implement API calls for updating and fetching order with it's details.
- Write unit tests to ensure the system works correctly.
#### How should this be manually tested?
- Ensure You have backend service running and set VITE_APP_API_URL in your .env file.
- Clone this repository and navigate to the project directory
  ```bash
  git clone -b ft-register https://github.com/atlp-rwanda/knights-ecomm-fe.git
  cd knights-ecomm-fe
  git checkout ft-order-management
  npm i
  npm run dev
  ```
#### What are the relevant pivotal tracker/Trello stories?
#68 

#### Screenshots (if appropriate)
![image](https://github.com/user-attachments/assets/1212d1bd-4e32-47db-a43b-a0fc6ea0b7c8)
![image](https://github.com/user-attachments/assets/2fbbb498-0060-4a83-9712-23a1b2e9f950)
